cmd/es-index-cleaner/app/index_client.go

```diff
@@ -0,0 +1,144 @@
+	// Index creation time.
+	CreationTime time.Time
+	// Aliases
+	Aliases map[string]bool
+}
+
+// IndicesClient is a client used to manipulate with indices.
+type IndicesClient struct {
+	// Http client.
+	Client *http.Client
+	// ES server endpoint.
+	Endpoint string
+	// ES master_timeout parameter.
+	MasterTimeoutSeconds int
+	BasicAuth            string
+}
+
+// GetJaegerIndices queries all Jaeger indices including the archive and rollover.
+// Jaeger daily indices are:
+//     jaeger-span-2019-01-01, jaeger-service-2019-01-01, jaeger-dependencies-2019-01-01
+//     jaeger-span-archive
+// Rollover indices:
+//     aliases: jaeger-span-read, jaeger-span-write, jaeger-service-read, jaeger-service-write
+//         indices: jaeger-span-000001, jaeger-service-000001 etc.
+//     aliases: jaeger-span-archive-read, jaeger-span-archive-write
+//         indices: jaeger-span-archive-000001
+func (i *IndicesClient) GetJaegerIndices(prefix string) ([]Index, error) {
+	prefix += "jaeger-*"
+	r, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/%s?flat_settings=true&filter_path=*.aliases,*.settings", i.Endpoint, prefix), nil)
+	if err != nil {
+		return nil, err
+	}
+	i.setAuthorization(r)
+	response, err := i.Client.Do(r)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query Jaeger indices: %q", err)
+	}
+
+	if response.StatusCode != 200 {
+		body, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("failed to query Jaeger indices: %s", string(body))
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	type indexInfo struct {
+		Aliases  map[string]interface{} `json:"aliases"`
+		Settings map[string]string      `json:"settings"`
+	}
+	var indicesInfo map[string]indexInfo
+	if err = json.Unmarshal(body, &indicesInfo); err != nil {
+		return nil, fmt.Errorf("failed to unmarshall response: %q", err)
+	}
+
+	var indices []Index
+	for k, v := range indicesInfo {
+		aliases := map[string]bool{}
+		for alias := range v.Aliases {
+			aliases[alias] = true
+		}
+		// ignoring error, ES should return valid date
+		creationDate, _ := strconv.ParseInt(v.Settings["index.creation_date"], 10, 64)
+
+		indices = append(indices, Index{
+			Index:        k,
+			CreationTime: time.Unix(0, int64(time.Millisecond)*creationDate),
+			Aliases:      aliases,
+		})
+	}
+	return indices, nil
+}
+
+// DeleteIndices deletes specified set of indices.
+func (i *IndicesClient) DeleteIndices(indices []Index) error {
+	concatIndices := ""
+	for _, i := range indices {
+		concatIndices += i.Index
+		concatIndices += ","
+	}
+
+	r, err := http.NewRequest(http.MethodDelete, fmt.Sprintf("%s/%s?master_timeout=%ds", i.Endpoint, concatIndices, i.MasterTimeoutSeconds), nil)
+	if err != nil {
+		return err
+	}
+	i.setAuthorization(r)
+
+	response, err := i.Client.Do(r)
+	if err != nil {
+		return err
+	}
+	if response.StatusCode != 200 {
+		var body string
+		if response.Body != nil {
```

Comment:
This logic seems to be similar to the logic for handling errors in https://github.com/jaegertracing/jaeger/pull/3192/files#diff-1980448a64404c77331625605cdeb179c82f7cbab71895fbf049ec68779c0a10R68  

Can we create a function for handle that logic? so don't need to repeat this.